### PR TITLE
🌱 Support for management v3 downgrade to provisioning v1 with automigration

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:

Add downgrade procedure as described in the ADR #579 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [x] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

File: charts/rancher-turtles/templates/post-upgrade-job.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if and (eq (index .Values "rancherTurtles" "features" "managementv3-cluster" "enabled") true) (eq (index .Values "rancherTurtles" "features" "managementv3-cluster-migration" "enabled") true) }}
```
Comment (furkatgofurov7) [inline charts/rancher-turtles/templates/post-upgrade-job.yaml:1]: why we do not need to enable `managementv3-cluster` feature flag anymore?

File: test/e2e/suites/migrate-gitops/suite_test.go [hunk 2]
```diff
@@ -179,133 +146,50 @@ var _ = BeforeSuite(func() {
...
-					Name:      "capa-controller-manager",
-					Namespace: "capa-system",
-				},
-				{
-					Name:      "capz-controller-manager",
-					Namespace: "capz-system",
-				},
-			},
-		})
-	} else if Label(e2e.LocalTestLabel).MatchesLabelFilter(GinkgoLabelFilter()) {
```
Comment (furkatgofurov7) [inline test/e2e/suites/migrate-gitops/suite_test.go:282]: I am not sure how vSphere tests are related to downgrade process? Can you please explain why tests here and above are removed completely.

Comment (Danil-Grigorev) [reply test/e2e/suites/migrate-gitops/suite_test.go:282]: They are not 🤷🏻

File: charts/rancher-turtles/templates/post-upgrade-job.yaml (hunk 1 above)
Comment (Danil-Grigorev) [reply charts/rancher-turtles/templates/post-upgrade-job.yaml:1]: Downgrade procedure involves `managementv3-cluster` being disabled. The desired behavior was explained in ADR.

**Review by alexander-demicev — APPROVED** [review]

**Review by salasberryfin — APPROVED** [review]

**Review by furkatgofurov7 — APPROVED** [review]

File: test/e2e/suites/migrate-gitops/suite_test.go (hunk 2 above)
Comment (furkatgofurov7) [reply test/e2e/suites/migrate-gitops/suite_test.go:282]: Probably copy paste mistake

Comment (Danil-Grigorev) [reply test/e2e/suites/migrate-gitops/suite_test.go:282]: There is been a discussion of refactoring e2e suite. You can find many of those across different suites. We need a `builder` for suite, which will allow us to construct:
- If it is AKS based execution
- Is VSphere involved
- Is this a local run
- Is it isolated execution using `sslip.io` or not
- Any additional modifications to network, providers, etc.

At this point it is a boilerplate which tends to be duplicated and unfortunately it is mandatory.

Comment (Danil-Grigorev) [reply test/e2e/suites/migrate-gitops/suite_test.go:282]: I opened an issue related to this discussion and previous mentions: https://github.com/rancher/turtles/issues/603

**Pull request merged**